### ISSUE TITLE
Asset Hub : assets/balances page takes a lot of time to load

### ISSUE THREAD

**Issue by muddlebee** [polkadot-js/apps#10476]
* **I'm submitting a ...**

  - [x] Bug report
  - [ ] Feature request
  - [ ] Support request
  - [ ] Other

* **What is the current behavior and expected behavior?**

 assets/balances at Polkadot Asset Hub just keeps loading, even with different RPCs. I noticed this common behaviour.

https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fstatemint-rpc-tn.dwellir.com#/assets/balances
![image](https://github.com/polkadot-js/apps/assets/8139783/8a080ee0-7d09-4971-8ff0-c1222bae342a)

* **What is the motivation for changing the behavior?**

Optimize the APIs/fetch functionality of assets/balances at Asset Hub.

* **Please tell us about your environment:**

  - Version: Live URL

**Comment by IkerAlus** [comment]
not able to reproduce, it took less than a second from several devices. 

I would say it was a temporary outage at the RPC provider side.

**Comment by muddlebee** [comment]
@IkerAlus specifically the "Balances" tab 

https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fstatemint-rpc-tn.dwellir.com#/assets/balances

![image](https://github.com/polkadot-js/apps/assets/8139783/f18e00d8-76e3-456d-a34f-ca42210c26d4)

**Comment by IkerAlus** [comment]
@muddlebee you are right, it is not that it keeps loading but it simply does not show any data even after selecting a specific asset to  query for balances: 
<img width="488" alt="Screenshot 2024-04-25 at 15 25 44" src="https://github.com/polkadot-js/apps/assets/34474035/6b393c25-b378-474a-82f4-c7ca67a60eba">

**Comment by MoMannn** [comment]
This is not as much of loading for a long time but straight bug:
<img width="1261" alt="Screenshot 2024-05-17 at 15 49 14" src="https://github.com/polkadot-js/apps/assets/8865298/ea18588d-ac2b-40e3-950d-e4c632cd927b">

**Comment by TarikGul** [comment]
@MoMannn How would i reproduce the issue: What chain did you go to, and what action did you try to perform?

**Comment by muddlebee** [comment]
@TarikGul check here https://github.com/polkadot-js/apps/issues/10476#issuecomment-2073146953

error is coming due to this https://github.com/polkadot-js/apps/blob/master/packages/page-assets/src/Balances/index.tsx#L24-L25

> const [info, setInfo] = useState<AssetInfoComplete | null>(null);
this is returning null so id is not populated in "Balances"

**Comment by muddlebee** [comment]
I  suspect it was introduced here https://github.com/polkadot-js/apps/pull/10423/files

**Comment by TarikGul** [comment]
@muddlebee Feel free to test out the solution in: https://github.com/polkadot-js/apps/pull/10592

**Comment by muddlebee** [comment]
@TarikGul still facing the issue and it keeps loading with some console errors

![image](https://github.com/polkadot-js/apps/assets/8139783/8dc57701-2b09-4ae0-b932-ae1b5333ba2e)

**Comment by TarikGul** [comment]
@muddlebee I'll have it fully fixed on Monday. But for now will reopen in.

**Comment by TarikGul** [comment]
So I get the same error in the browser, but I don't have a loading issue oddly enough. 

That being said what is so weird about this error is that it doesn't persist in in dev mode.

**Comment by ap211unitech** [comment]
This issue got fixed with this [PR](https://github.com/polkadot-js/apps/pull/11222)
